sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -234,16 +238,24 @@ private[spark] object CosmosConfigNames {
 }
 
 private object CosmosConfig  extends BasicLoggingTrait {
-  lazy val accountDataResolverCls = getAccountDataResolver
-  def getAccountDataResolver(): Option[AccountDataResolver] = {
-    logInfo("Checking for account resolvers")
+  def getAccountDataResolver(config: Map[String, String]): Option[AccountDataResolver] = {
+    val accountDataResolverServiceName : Option[String] = config.get(CosmosConfigNames.AccountDataResolverServiceName)
+
+    logInfo(s"Checking for account resolvers - requested service name '${accountDataResolverServiceName.getOrElse("n/a")}'")
     var accountDataResolverCls = None: Option[AccountDataResolver]
     val serviceLoader = ServiceLoader.load(classOf[AccountDataResolver])
     val iterator = serviceLoader.iterator()
-    if (iterator.hasNext()) {
+    while (!accountDataResolverCls.isDefined && iterator.hasNext()) {
```

Comment:
why do we need a while loop here? to support multiple?

Reply:
Yes - when multiple services implementing AccountDataResolver are on the classpath. The new config allows specifying which one to use in that case (instead of always using the random first) - Synpase (for linked service) + custom account data resolver would make this necessary.